en/docs/chapter_sorting/counting_sort.md

```diff
@@ -24,18 +24,18 @@ The code is shown below:
 
 ## Complete implementation
 
-Astute readers might have noticed, **if the input data is an object, the above step `3.` becomes ineffective**. Suppose the input data is a product object, we want to sort the products by their price (a class member variable), but the above algorithm can only provide the sorting result for the price.
+Careful readers might have noticed, **if the input data is an object, the above step `3.` is invalid**. Suppose the input data is a product object, we want to sort the products by the price (a class member variable), but the above algorithm can only give the sorted price as the result.
```

Comment:
"the" in "sort the products by the price" is not necessary.

Reply:
I mean the second "the" before "price".

Reply:
I think it's still necessary. "give the sorted price". If you want to remove "the", "sorted price" should be considered as a proper noun, like "SORTED_PRICE"